### DISCUSSION TITLE
Extending Canvas

### DISCUSSION THREAD

**Discussion by Biome21** [ImageMagick/ImageMagick#4139] [Help] [answered]
Hi,So im trying to read a psb file and setting the canvas bound of each layer to encompass the other layer. Basically iterating thru all the IMagickImage of the MagickImageCollection (the psb). For each image doing another for loop against the other IMagickImage and then do collection[i].Extent(collection[j]).

`MagickImageCollection psbFile = new MagickImageCollection(fileToImportPath);`
`for (int i = 0; i < psbFile.Count; i++) {`
`            for (int j = i + 1; j < psbFile.Count; j++)
            {
                psbFile[i].Extent(psbFile[j].Page);
            }
}`

However if I then try to save them on disk using write. All my pngs are solid white.
What im trying to achieve is lets say I have these 2 images (A and B) in the collection, I want both image to encompass the other preserving the position in the final canvas (so for example A stays top left and B stays bottom right. 
![Screen Shot 2021-08-31 at 4 20 02 PM](https://user-images.githubusercontent.com/47017071/131570697-c9efff31-8e1f-4aa0-913d-cf63d9837ca2.png)

Disclaimer : Im using Magick.Net dll and not the commandline

thank you very much

**Comment by fmw42** [comment]
Post your PSB file.

**Reply by Biome21** [reply]
Here's a small example
[HappyAngry.psb.zip](https://github.com/ImageMagick/ImageMagick/files/7086757/HappyAngry.psb.zip)

**Comment by fmw42** [comment]
This is how to do it in command line. Sorry, I do not know Magick.NET.

```
convert HappyAngry.psb[0] -alpha transparent HappyAngry.psb[1] -layers flatten x1.png
convert HappyAngry.psb[0] -alpha transparent HappyAngry.psb[2] -layer2 flatten x2.png
```

**Comment by dlemstra** [comment]
Do you still need help translating this to Magick.NET @Biome21?

**Comment by Biome21** [comment, marked as answer]
So I was able to fix my problem by doing it in 2 steps
1) Creating a new MagickImage that would be my total canvas. i used MagickImageCollection.Merge
2) Iterate over all the layers of the MagickImageCollection and calling Extent on each layer with the dimensions of the mergedImage
Something like that
`MagickImageCollection psbFile = new MagickImageCollection(fileToImportPath);`
`var mergedImage = new MagickImage(psbFile.Merge()))`
`for (int i = 0; i < psbFile.Count; i++)`
 `{`
             `//Since in photoshop all positions are calculated from top left`
             `MagickGeometry canvas = new MagickGeometry(mergedImage.Page.X - psbFile[i].Page.X,`
                                                                                                  `mergedImage.Page.Y - psbFile[i].Page.Y,`
                                                                                                  `mergedImageWidth,`
                                                                                                  `mergedImage.Height);`
            `psbFile[i].Extent(canvas, Gravity.Northwest, MagickColors.Transparent);`
